pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -338,11 +338,24 @@ func getConfig(ctx *types.SystemContext) (*V2RegistriesConf, error) {
 	configPath := ConfigPath(ctx)
 
 	configMutex.Lock()
-	defer configMutex.Unlock()
 	// if the config has already been loaded, return the cached registries
 	if config, inCache := configCache[configPath]; inCache {
+		configMutex.Unlock()
 		return config, nil
 	}
+	configMutex.Unlock()
+
+	return TryUpdatingCache(ctx)
+}
+
+// TryUpdatingCache loads the registries from the provided
+// `SystemContext` without using the internal cache. On success, the found
+// registries will be added into the internal registry cache.
+func TryUpdatingCache(ctx *types.SystemContext) (*V2RegistriesConf, error) {
```

Comment:
`s/the registries/the configuration/; s/the found registries/the loaded configuration/`, please.

LGTM otherwise (assuming tests pass).

Reply:
Changed as suggested :)